### PR TITLE
MUI controller lacks JSON response, instead returns status 200 with empty body

### PR DESCRIPTION
**Pull request by woutersamaey**
### Description (*)
Example URL http://hostname/admin/mui/index/render/key/xxx/?namespace=notification_area&sorting%5Bfield%5D=created_at&sorting%5Bdirection%5D=asc&isAjax=true

Can return status 200 with an empty body, when it should return JSON. This later causes a JavaScript error because data.items does not exist.

This happens when the admin user does not have enough ACL rights to this controller.

### Fixed Issues (if relevant)
1. Valid JSON response 
2. Correct HTTP status code

### Manual testing scenarios (*)
1. Create an admin with very limited access rights (only orders for example)
2. Go to the orders page.
3. Look for error

data-storage.js:270 Uncaught TypeError: Cannot read property 'items' of null
    at UiClass.onRequestComplete (data-storage.js:270)
    at fire (jquery.js:3232)
    at Object.fireWith [as resolveWith] (jquery.js:3362)
    at done (jquery.js:9840)
    at XMLHttpRequest.callback (jquery.js:10311)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @woutersamaey. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by woutersamaey** [comment]
Forgot the return line at the end. Added it to the PR. Sorry.

**Comment by sidolov** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @sidolov. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @sidolov, here is your [Magento instance](https://i-19859-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19859-2-3-develop.instances.magento-community.engineering/admin](https://i-19859-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sivaschenko** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, here is your [new Magento instance](https://pr-19859.instances.magento-community.engineering).
Admin access: [https://pr-19859.instances.magento-community.engineering/admin](https://pr-19859.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by sivaschenko** [comment]
Hi @woutersamaey thanks for your contribution! Could you please fix the code style for static tests to pass.

With your fix an admin with limited permissions will see the screen like:
image

It would be nice to have an appropriate error (Not enough permissions), however, that's probably out of this pull request scope.

**Comment by woutersamaey** [comment]
@sivaschenko You're right about this error message and indeed, it is out of scope. This error can come up in a number of situations and it's always vague. A more thorough refactoring is needed.

Also, the admin notifications should not try to load in the first place, as the user doesn't have the required rights, so dealing with the root cause is best.

These could be 2 new separate issues.

About the code style, this is just another one of those "i dont like else-statements" messages. We've ignored those in the past due to an overzealous Codacy.

File: app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php [hunk 1]
```diff
@@ -86,6 +86,18 @@ public function execute()
 
                 $contentType = $this->contentTypeResolver->resolve($component->getContext());
                 $this->getResponse()->setHeader('Content-Type', $contentType, true);
+            }else {
```
Comment (sivaschenko) [inline app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php]:
```suggestion
            } else {
```

File: app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php [hunk 2]
```diff
@@ -86,6 +86,18 @@ public function execute()
...
+                /** @var \Magento\Framework\Controller\Result\Json $resultJson */
+                $resultJson = $this->resultJsonFactory->create();
+                $resultJson->setStatusHeader(
+                    \Zend\Http\Response::STATUS_CODE_403,
+                    \Zend\Http\AbstractMessage::VERSION_11,
+                    'Forbidden'
+                );
+                return $resultJson->setData([
+                        'error' => $this->escaper->escapeHtml('Forbidden'),
+                        'errorcode' => 403]
```
Comment (sivaschenko) [inline app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php]:
```suggestion
                        'errorcode' => 403
```

File: app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php [hunk 3]
```diff
@@ -86,6 +86,18 @@ public function execute()
...
+                );
```
Comment (sivaschenko) [inline app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php]:
```suggestion
                ]);
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @woutersamaey thanks for your reply. We will not be able to merge the pull request to mainline without code style fixes. Please see my code suggestions. All you need to to do is to add these suggestions to batch and then commit them using github UI.

**Comment by woutersamaey** [comment]
@sivaschenko OK, now I see what you mean. Thanks for preparing this for me. I have commited your suggestions.

Is everything OK now? Honestly I'm not 100% familiar with Github so please excuse me if I didn't understand correctly.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3725](https://jira.corp.magento.com/browse/ENGCOM-3725) has been created to process this Pull Request

**Pull request merged**